dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -371,8 +406,12 @@ public boolean hasNext() throws HpccFileException
      */
     public Object getNext() throws HpccFileException
     {
-        long start = System.nanoTime();
-        this.externalProcessingTimeNS += (start - lastRecordTimeNS);
+        long start = 0;
+        if (this.recordBuilderSpan != null)
```

Comment:
Have we established a simple way determine if tracing is currently not enabled? I'm concerned that this.recordBuilderSpan could be instantiated, but not valid/recording. In those situations, we should avoid gathering timing information for performance reasons.